### ISSUE TITLE
FallbackFactory for with Feign does not works

### ISSUE THREAD

**Issue by kerwin612** [resilience4j/resilience4j#2000]
Resilience4j version: v2.1.0

Java version: 17

This issue can be referred to #560 
Unlike the 560, I use: `withFallbackFactory` 

The test environment is as follows:  

**pom.xml**  
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>3.1.1</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>
	<groupId>com.example</groupId>
	<artifactId>test-for-resilience4j</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<name>test-for-resilience4j</name>
	<description>Test for resilience4j</description>
	<properties>
		<java.version>17</java.version>
	</properties>
	<dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <dependency>
            <groupId>io.github.resilience4j</groupId>
            <artifactId>resilience4j-feign</artifactId>
            <version>2.1.0</version>
        </dependency>
		<dependency>
			<groupId>io.github.openfeign</groupId>
			<artifactId>feign-core</artifactId>
            <version>12.2</version>
		</dependency>
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>4.13.2</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.assertj</groupId>
            <artifactId>assertj-core</artifactId>
            <version>3.24.2</version>
            <scope>test</scope>
        </dependency>
		<dependency>
			<groupId>com.github.tomakehurst</groupId>
			<artifactId>wiremock-jre8-standalone</artifactId>
            <version>2.35.0</version>
            <scope>test</scope>
		</dependency>
	</dependencies>

</project>
```

**tests**
```java
package com.example.test;


import feign.RequestLine;
import io.github.resilience4j.feign.FeignDecorators;
import io.github.resilience4j.feign.Resilience4jFeign;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;

import com.github.tomakehurst.wiremock.junit.WireMockRule;

import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
import static com.github.tomakehurst.wiremock.client.WireMock.get;
import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
import static com.github.tomakehurst.wiremock.client.WireMock.verify;
import static org.assertj.core.api.Assertions.assertThat;

public class Tests {

    private static final String MOCK_URL = "http://localhost:8080/";

    @Rule
    public WireMockRule wireMockRule = new WireMockRule();

    private TestService testService;

    @Before
    public void setUp() {
        final FeignDecorators decorators = FeignDecorators.builder()
            .withFallbackFactory(e -> createLambdaFallback())
            .build();

        this.testService = Resilience4jFeign.builder(decorators)
            .target(TestService.class, MOCK_URL);
    }

    @Test
    public void testFallback() {
        setupStub();

        final String result = testService.greeting();

        assertThat(result).describedAs("Result").isEqualTo("fallback");
        verify(1, getRequestedFor(urlPathEqualTo("/greeting")));
    }

    private void setupStub() {
        stubFor(get(urlPathEqualTo("/greeting"))
            .willReturn(aResponse()
                .withStatus(400)
                .withHeader("Content-Type", "text/plain")
                .withBody("Hello, world!")));
    }

    public static TestService createLambdaFallback() {
        return () -> "fallback";
    }

    public interface TestService {

        @RequestLine("GET /greeting")
        String greeting();
    }

}
```  
The full code is at: [test-for-resilience4j](https://github.com/kerwin612/resilience4j/tree/test-for-resilience4j)

**Comment by Eyesun23** [comment]
@RobWin will not solve the issue but only ask questions... I feel like this library is not being monitored by anyone

**Comment by kerwin612** [comment]
Fixed in #1999
